subprojects/core/src/main/java/org/gradle/api/internal/DefaultNamedDomainObjectCollection.java

```diff
@@ -103,8 +103,18 @@ protected DefaultNamedDomainObjectCollection(Class<? extends T> type, ElementSou
     }
 
     // should be protected, but use of the class generator forces it to be public
-    public DefaultNamedDomainObjectCollection(DefaultNamedDomainObjectCollection<? super T> collection, CollectionFilter<T> filter, Instantiator instantiator, Namer<? super T> namer) {
-        this(filter.getType(), collection.filteredStore(filter), collection.filteredEvents(filter), collection.filteredIndex(filter), instantiator, namer);
+    public DefaultNamedDomainObjectCollection(DefaultNamedDomainObjectCollection<? super T> collection, CollectionFilter<T> elementFilter, Instantiator instantiator, Namer<? super T> namer) {
+        this(collection, null, elementFilter, instantiator, namer);
+    }
+
+    protected DefaultNamedDomainObjectCollection(
+        DefaultNamedDomainObjectCollection<? super T> collection,
+        @Nullable Spec<String> nameFilter,
```

Comment:
❓ Wouldn't it be simpler if `nameFilter` was not nullable but use `Specs.SATISFIES_ALL` as the default instead?

Reply:
Yes, it would! I will make the change.